tracing-test-utils/src/main/java/com/palantir/tracing/TestTracingExtension.java

```diff
@@ -39,12 +40,11 @@
 final class TestTracingExtension implements BeforeTestExecutionCallback, AfterTestExecutionCallback {
 
     private static final Logger log = LoggerFactory.getLogger(TestTracingExtension.class);
-    private final TestTracingSubscriber subscriber = new TestTracingSubscriber();
```

Comment:
This should be kept as a non-final field that we reset before every test. Then we don't need to cast.

Reply:
Sorry, that's not quite right. This needs to be stored in the `ExtensionContext`.

https://junit.org/junit5/docs/current/user-guide/#extensions-keeping-state